src/planet-a/IcoSphereTree.h

```diff
@@ -67,8 +81,70 @@ static constexpr int gc_icosahedronVertCount = 12;
 static constexpr std::uint8_t gc_triangleMaskSubdivided = 0b0001;
 //static constexpr std::uint8_t gc_triangleMaskChunked    = 0b0010;
 
-// An icosahedron with subdividable faces
-// it starts with 20 triangles, and each face can be subdivided into 4 more
+/**
+ * A simple 1D translation and scale
+ */
+struct TriangleSideTransform
+{
+    float m_translation{0.0f};
+    float m_scale{1.0f};
+};
+
+class IcoSphereTreeObserver
+{
+public:
+    using Handle_T = std::list<std::weak_ptr<IcoSphereTreeObserver>>::iterator;
+
+    virtual void on_ico_triangles_added(std::vector<trindex_t> const&) = 0;
```

Comment:
This looks like a good spot for std::span, but sadly we can't use that yet, since C++20 isn't really available anywhere.

Reply:
cppreference says it's available on all of GCC libstdc++, Clang libc++, MSVC, and Apple Clang. Not sufficient?

Reply:
Is it available in Ubuntu 20.04?

Additionally, turning on C++20 also implies that people can use other C++20 features.

I think it would be better to wait a few more months until most people's dev machines have C++20 available. Even on my gentoo computers, I can't use C++20 yet. Gentoo's still on GCC 9 because GCC changed a default and fucked a bunch of packages up.

Reply:
I'm not sure, I've only tested C++20 features on MSVC. We can wait though if the latest gcc isn't available everywhere yet.